.gitignore

```diff
@@ -1,5 +1,4 @@
 /target
-Cargo.lock
```

Comment:
Why?

Reply:
https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html

The motivation could be to make the CI more stable. Though it is worth thinking about if we should add `cargo update` to one of the CI flows so that we can know if a newer version of a dependency causes the library to break.

Reply:
Thank you for the comment, removed